backend/src/Altinn.Authorizationadmin/Altinn.Authorizationadmin/Services/Implementation/PolicyAdministrationPoint.cs

```diff
@@ -143,25 +149,31 @@ private async Task<bool> WriteDelegationPolicyInternal(string policyPath, List<R
 
             if (!string.IsNullOrWhiteSpace(resourceRegistryId))
             {
-                XacmlPolicy appPolicy = await _prp.GetPolicyAsync(resourceRegistryId);
-                if (appPolicy == null)
+                XacmlPolicy resourcePolicy = await _prp.GetPolicyAsync(resourceRegistryId);
+                if (resourcePolicy == null)
                 {
-                    _logger.LogWarning("No valid App policy found for delegation policy path: {policyPath}", policyPath);
+                    _logger.LogWarning("No valid resource policy found for delegation policy path: {policyPath}", policyPath);
                     return false;
                 }
 
-                //if (!DelegationHelper.CheckIfPolicyContainsMatchingRule(appPolicy, rules, out PolicyRule rule))
-                //{
-                //    _logger.LogWarning("Matching rule not found in app policy. Action might not exist for Resource, or Resource itself might not exist. Delegation policy path: {policyPath}. Rule: {rule}", policyPath, rule);
-                //    return false;
-                //}
                 foreach (Rule rule in rules)
                 {
-                    if (!DelegationHelper.PolicyContainsMatchingRule(appPolicy, rule))
+                    if (!DelegationHelper.PolicyContainsMatchingRule(resourcePolicy, rule))
                     {
-                        _logger.LogWarning("Matching rule not found in app policy. Action might not exist for Resource, or Resource itself might not exist. Delegation policy path: {policyPath}. Rule: {rule}", policyPath, rule);
+                        _logger.LogWarning("Matching rule not found in resource policy. Action might not exist for Resource, or Resource itself might not exist. Delegation policy path: {policyPath}. Rule: {rule}", policyPath, rule);
                         return false;
                     }
+
+                    ServiceResource resource = new ServiceResource();
```

Comment:
All rules is already sorted into sets for delegations of the same resource (the resourceRegistryId variable).
So shouldnt have to iterate over all rules internally here to check each against _resourceRegistryClient.GetResource.

Instead this can be performed once along side verification that the resourcePolicy exist